### PR TITLE
ISO 19115 CLI

### PR DESCRIPTION
**Pull request by AlexDo1**
In this draft Pull Request, the (not working) example of registering a CLI command via an extension is implemented.

When running the CLI (e.g. `metacatalog -h`) the following error is produced:
```
Traceback (most recent call last):
  File "/home/alexander/anaconda3/bin/metacatalog", line 33, in <module>
    sys.exit(load_entry_point('metacatalog', 'console_scripts', 'metacatalog')())
  File "/home/alexander/Github/metacatalog/metacatalog/command_line.py", line 105, in main
    extension.init_cli(subparsers=subparsers, defaults=default_options)
TypeError: init_cli() got multiple values for argument 'subparsers'
```

### CODE REVIEW

File: metacatalog/command_line.py [hunk 1]
```diff
@@ -98,6 +98,12 @@ def main():
     migration_parser.add_argument('--message', '-m', type=str, help="Optional revision message. Only used for revision command")
     migration_parser.set_defaults(func=migrate)
 
+    # check extensions for CLI registers
+    from metacatalog.ext import EXTENSIONS
+    for extension in EXTENSIONS.values():
+        if hasattr(extension, 'init_cli'):
```
Comment (AlexDo1) [inline metacatalog/command_line.py:104]: This always evaluates `True`, as class `MetacatalogExtensionInterface` always has the classmethod `init_cli()`.  
I changed the line to `if hasattr(extension, 'cli')`, but this does not resolve the error.

Comment (mmaelicke) [reply metacatalog/command_line.py:104]: Yeah, that was intended behavior, as the `init_cli` is **not** an `ab.cabstractclassmethod`. Thus, you can implement an interface, that does not have a `init_cli` method.
A `cli` method is not a thing, and btw. not a very useful name.

File: metacatalog/ext/standard_export/extension.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        wrapper_entry.__doc__ = StandardsExportExtension.iso19115_export.__doc__
+        wrapper_entry.__name__ = StandardsExportExtension.iso19115_export.__name__
+        
+        # add wrapper to Entry model
+        Entry.export_iso19115 = wrapper_entry
+
+    @classmethod
+    def cli(cls, args):
+        if not args.quiet:
+            print(args.foo.upper())
```
Comment (mmaelicke) [inline metacatalog/ext/standard_export/extension.py:48]: This should definitely have a useful name.

File: metacatalog/ext/standard_export/extension.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    @classmethod
+    def init_cli(subparsers, defaults):
```
Comment (mmaelicke) [inline metacatalog/ext/standard_export/extension.py:51]: This is causing the error, as you miss the `cls` argument, that any `@classmethod` always has to accept. Your method accepts the passed Extension class as subparsers argument and then a second subparsers argument is passed as a named argument. That is the error.

**Review by mmaelicke — CHANGES_REQUESTED** [review]
We have to keep in mind that #269 should be merged before this one.

File: metacatalog/ext/standard_export/extension.py (hunk 2 above)
Comment (AlexDo1) [reply metacatalog/ext/standard_export/extension.py:48]: Sure, I just wanted to try out your example.  
Now that registering a new command works, I can start from here.

File: metacatalog/ext/standard_export/extension.py (hunk 3 above)
Comment (AlexDo1) [reply metacatalog/ext/standard_export/extension.py:51]: That makes sense, the error is solved, thanks!

**Comment by AlexDo1** [comment]
I will open a new branch from updated main for the ISO CLI feature.

**Pull request closed without merging**